compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -115,52 +121,77 @@ LogicalResult distributeLocalMemory(ModuleOp moduleOp) {
         newDeallocOp->moveBefore(&newAllocOp->getBlock()->back());
         memrefToNew[allocOp] = newAllocOp;
       }
-      auto newAlloc = memrefToNew[allocOp];
-      rewriter.replaceAllUsesWith(subviewOp, newAlloc);
-      toBeErased.push_back(subviewOp);
     }
 
-    // Update the alloc's DMA users.
-    if (memrefToNew.contains(allocOp)) {
-      LLVM_DEBUG(llvm::dbgs()
-                 << "Update allocate DMA users: " << dmaUsers.size() << "\n");
-      auto newAlloc = memrefToNew[allocOp];
-      auto type = cast<MemRefType>(newAlloc.getType());
-      for (AMDAIE::DmaCpyNdOp dmaOp : dmaUsers) {
-        SmallVector<Value> empty;
-        rewriter.setInsertionPoint(dmaOp.getSourceObjectFifo());
-        auto source = rewriter.create<AMDAIE::LogicalObjectFifoFromMemrefOp>(
-            rewriter.getUnknownLoc(), LogicalObjectFifoType::get(type),
-            newAlloc.getResult());
-        rewriter.replaceOp(dmaOp.getSourceObjectFifo(), source);
-        rewriter.setInsertionPoint(dmaOp);
-        auto newDmaOp = rewriter.create<AMDAIE::DmaCpyNdOp>(
-            dmaOp.getLoc(), dmaOp.getTarget(), dmaOp.getTargetMixedOffsets(),
-            dmaOp.getTargetMixedSizes(), dmaOp.getTargetMixedStrides(), source,
-            dmaOp.getSourceMixedOffsets(), dmaOp.getSourceMixedSizes(),
-            dmaOp.getSourceMixedStrides());
-        rewriter.replaceOp(dmaOp, newDmaOp);
-        // We have to discard non-zero offsets as subview has been replaced by a
-        // dedicated allocated memref.
-        SmallVector<int64_t> allocShape(type.getShape());
-        (void)discardAllNonZeroOffsets<CopyOpOperateOn::Source>(
-            rewriter,
-            cast<AMDAIE::DoublyStridedOpInterface>(newDmaOp.getOperation()),
-            allocShape);
+    if (!memrefToNew.contains(allocOp)) return WalkResult::advance();
+
+    // Traverse the following users of the AllocOp and replace uses of current
+    // AllocOp within them by a new AllocOp :-
+    // 1. vector.transfer_read.
+    // 2. vector.transfer_write.
+    // 3. memref.subview - only for this op we would replace its uses with a new
```

Comment:
You also replace the uses for `vector::TransferReadOp`: https://github.com/nod-ai/iree-amd-aie/pull/493/files#diff-1bb5dc575fe681a4b367fc1f2515f08c6361b6b22c98bd7ef50a88efaed49cdbR145 ?

Reply:
I've updated the comment to reflect what I'm doing better now. Could you check now?

Reply:
I see, thanks, this clarification is helpful